scheduler/replica_scheduler.go

```diff
@@ -574,42 +600,23 @@ func (rcs *ReplicaScheduler) RequireNewReplica(replicas map[string]*longhorn.Rep
 		return 0
 	}
 
-	// Otherwise Longhorn will relay the new replica creation then there is a chance to reuse failed replicas later.
-	settingValue, err := rcs.ds.GetSettingAsInt(types.SettingNameReplicaReplenishmentWaitInterval)
+	timeUntilNext, timeOfNext, err := rcs.timeToReplacementReplica(volume)
 	if err != nil {
-		logrus.Errorf("Failed to get Setting ReplicaReplenishmentWaitInterval, will directly replenish a new replica: %v", err)
-		return 0
+		msg := "Failed to get time until replica replacement, will directly replenish a new replica"
+		logrus.WithError(err).Errorf(msg)
 	}
-	waitInterval := time.Duration(settingValue) * time.Second
-	lastDegradedAt, err := util.ParseTime(volume.Status.LastDegradedAt)
-
-	if err != nil {
-		logrus.Errorf("Failed to get parse volume last degraded timestamp %v, will directly replenish a new replica: %v", volume.Status.LastDegradedAt, err)
-		return 0
-	}
-	now := time.Now()
-	if now.After(lastDegradedAt.Add(waitInterval)) {
-		return 0
+	if timeUntilNext > 0 {
+		logrus.Infof("Replica replenishment is delayed until %v", timeOfNext)
 	}
-
-	logrus.Infof("Replica replenishment is delayed until %v", lastDegradedAt.Add(waitInterval))
-	// Adding 1 more second to the check back interval to avoid clock skew
-	return lastDegradedAt.Add(waitInterval).Sub(now) + time.Second
+	return timeUntilNext
```

Comment:
Yikes, I think you are right. If so, that will be pretty ugly. I'll test it an modify accordingly if true.

Reply:
It was actually not possible to have a negative value returned from `timeToReplacementReplica` when called in `RequireNewReplica` because of the redundant check for degraded (we would never have reached `timeToReplacementReplica` if `timeToReplacementReplica` would have returned a -1.

In light of this, I just removed the check from `timeToReplacementReplica` to simplify it and added a similar check in `getDiskCandidates`.

Reply:
Clearer, thanks!